packages/opentelemetry-core/src/platform/browser/environment.ts

```diff
@@ -25,7 +25,7 @@ import {
  * Gets the environment variables
  */
 export function getEnv(): Required<ENVIRONMENT> {
-  const _window = window as typeof window & RAW_ENVIRONMENT;
-  const globalEnv = parseEnvironment(_window);
+  const _global = (typeof globalThis === 'object' ? globalThis : window) as typeof globalThis & RAW_ENVIRONMENT;
```

Comment:
Non-Blocking comment as I'm going to be generally unavailable for 2 wks.

If we want to support all JS runtimes then there is more than just ```window``` and ```globalThis```.

There is also ```global``` and ```self``` we use [this helper for ApplicationInsights to abstract the global objects](https://github.com/microsoft/ApplicationInsights-JS/blob/master/tools/shims/src/Helpers.ts#L9-L39). OpenTelemetry should possible consider something similar.

Reply:
Generally, `globalThis`, defined in ECMAScript spec, should work fine and be the most preferred name to get the global object. The `window` here is just a fallback for browsers. 

It would be fine if we are going to use a set of fallbacks like `global` or `self`. Just like to mention that we have a couple of places that duplicate the same code piece yet they are not exported so we have to duplicate it again here:

- https://github.com/open-telemetry/opentelemetry-js-api/blob/main/src/platform/browser/globalThis.ts
- https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-api-metrics/src/platform/browser/globalThis.ts

Reply:
the metrics API will eventually be merged with the API. If the SDK is in need of a helper like this, `core` is the perfect place to define it.